pkg/tasks/prerequisites.go

```diff
@@ -111,6 +111,28 @@ func createEnvironmentFile(s *state.State) error {
 	return err
 }
 
+func disableNMCloudSetup(s *state.State, node *kubeoneapi.HostConfig, _ ssh.Connection) error {
+	if node.OperatingSystem == kubeoneapi.OperatingSystemNameRHEL &&
+		s.Cluster.CloudProvider.AWS != nil {
+		var allHosts = s.LiveCluster.ControlPlane
+		allHosts = append(allHosts, s.LiveCluster.StaticWorkers...)
+		for _, node := range allHosts {
+			if !node.Initialized() {
+				cmd, err := scripts.DisableNMCloudSetup()
+				if err != nil {
+					return err
+				}
+
+				_, _, err = s.Runner.RunRaw(cmd)
+
+				return err
+			}
+		}
+	}
```

Comment:
I'm pretty sure this is not what you want. This function is executing provided commands only on one specific node, which is provided via the `node` parameter.

You are going through all nodes, and for each node that's not initialized, run commands only on that one node. With this approach, if you have 2 initialized nodes and 1 uninitialized, you're going to run commands on initialized nodes as well.

What you need to do is to match the node you receive as a parameter to the node from `allHosts` array, and check if only that node is provisioned. If it isn't, run the commands and that's it.

Also, similar to the earlier comment, why don't we run this for all providers?

Reply:
> check if only that node is provisioned 

Provisioned or Initialized ? 

> Also, similar to the earlier comment, why don't we run this for all providers? 
because this service runs only on aws. We can however enable it for cloud providers